datafusion/core/src/datasource/listing/table.rs

```diff
@@ -147,18 +150,31 @@ impl ListingTableConfig {
             .await
             .ok_or_else(|| DataFusionError::Internal("No files for table".into()))??;
 
-        let file_extension =
-            ListingTableConfig::infer_file_extension(file.location.as_ref())?;
+        let (file_extension, maybe_compression_type) =
+            ListingTableConfig::infer_file_extension_and_compression_type(
+                file.location.as_ref(),
+            )?;
+
+        let mut format_options = HashMap::new();
+
+        let listing_file_extension =
+            if let Some(compression_type) = maybe_compression_type {
+                format_options
```

Comment:
Very minor: I think if you reoranized this code you could avoid `clone`ing the compression_type and file_extension.

Reply:
You're correct. Restructuring has been comitted.